#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QStackedWidget>
#include <QLineEdit>
#include <QLabel>
#include <QPushButton>
#include "gameboard.h"

namespace Ui {
class MainWindow;
}
class MainWindow : public QMainWindow {
    Q_OBJECT

public:
    explicit MainWindow(QWidget *parent = nullptr);
    ~MainWindow();
    static std::string generateRandomString(size_t length);

private slots:
    void onLoginButtonClicked();

private:
    Ui::MainWindow *ui;
    GameBoard *gameBoard;

};

#endif
